src/Microsoft.Framework.Runtime.Compilation.DesignTime/DesignTimeHostCompiler.cs

```diff
@@ -76,8 +76,7 @@ private void OnError(int? contextId, string error)
             int contextId;
             if (!contexts.TryGetValue(projectPath, out contextId))
             {
-                // This should never happen
-                throw new InvalidOperationException();
+                throw new InvalidOperationException($"The context of the project at {projectPath} could not be determined. It could be caused by multiple global.json files specifying different versions of the SDK for the same project.");
```

Comment:
How about: `$"The context of the project at {projectPath} could not be determined. This can happen if a project references other projects by source, and those projects have a global.json file specifying a different version of the SDK.`